clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -281,22 +284,23 @@ private boolean renameDirectory(Path src, Path dst) throws IOException {
             // May be unnecessary with https://github.com/treeverse/lakeFS/issues/1691
             LakeFSFileStatus dstFileStatus = getFileStatus(dst);
             if (!dstFileStatus.isDirectory()) {
-                // Same as https://github.com/apache/hadoop/blob/2960d83c255a00a549f8809882cd3b73a6266b6d/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java#L1527
-                throw new FileAlreadyExistsException("Failed rename " + src + " to " + dst
-                        + "; source is a directory and dest is a file");
+                // Same as https://github.com/apache/hadoop/blob/branch-2.7/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java#L482
+                LOG.debug("renameDirectory: rename src {} to dst {}: src is a directory and dst is a file", src, dst);
```

Comment:
From exception to debug logs... what a difference a major version makes...

Reply:
Throwing our problems or logging them